### DISCUSSION TITLE
Add is_holiday to csv

### DISCUSSION THREAD

**Discussion by yzorg** [gerrymanoim/exchange_calendars#267] [Ideas]
I help maintain trading systems and our hours & holiday calendars only support North America and London (XTSE, NYSE, OTC, NASDAQ, ARCA, XMEX, XLON).  I'm looking at expanding that to international this year.  This project looks great.

We render out each exchange trading hours much like the test CSVs, but ours also have `is_holiday` (start_time is null, end_time is null, but is_holiday =1).  This gives us performant join in SQL for orders that came in on a holiday.

All the North America (and Europe for that matter) follow Mon-Fri, so could use a negative test to know which are holidays.  But doesn't cover Middle East (XTAE) that aren't Mon-Fri.

So I searched a bit and found `weekmask(self)` on some calendars to know which are not Mon-Fri week schedules.  But I don't see any samples for how to access that via any documented APIs.  

I'm not doing historical, but in case others are historical would method differ for old or temporary changes to weekmask?

I'll keep digging into how the sessions are generated to try to figure this out myself, but thought I'd ask in case others have already solved it.

**Comment by yzorg** [comment]
I think I found it.  https://github.com/gerrymanoim/exchange_calendars/blob/master/docs/tutorials/calendar_properties.ipynb `In[13]` block says `nys.day` returns `CustomBusinessDay` type, which has the `nys.day.weekmask` property.  "1111100" is the default Mon-Fri schedule.

**Comment by maread99** [comment]
Hi @yzorg, sorry for not replying earlier.

Might the calendar's holidays properties serve your needs, `.adhoc_holidays` and `.regular_holidays`? The latter returns a pandas `AbstractHolidayCalendar` which would need to be queried, although it would be fairly simply to write an `is_holiday` method to query if a date is a holiday for a specific calendar (would be a useful addition to the ExchangeCalendar class).
